projection/core/src/main/scala/com/lightbend/lagom/internal/projection/ProjectionRegistryActor.scala

```diff
@@ -4,128 +4,186 @@
+  replicator ! Subscribe(ObservedStatusDataKey, self)
+  replicator ! Subscribe(NameIndexDataKey, self)
+
+  // (b) Keep a local copy to simplify the implementation of some ops
+  var requestedStatusLocalCopy: Map[WorkerKey, Status] = Map.empty[WorkerKey, Status]
+  var observedStatusLocalCopy: Map[WorkerKey, Status]  = Map.empty[WorkerKey, Status]
+
+  var nameIndexLocalCopy: Map[WorkerKey, WorkerCoordinates] = Map.empty[WorkerKey, WorkerCoordinates]
+
+  // (c) Actor indices contain only data of workers running locally
+  var actorIndex: Map[WorkerKey, ActorRef] = Map.empty[WorkerKey, ActorRef]
   // required to handle Terminate(deadActor)
-  var actorReverseIndex: Map[ActorRef, WorkerMetadata] = Map.empty[ActorRef, WorkerMetadata]
+  var reversedActorIndex: Map[ActorRef, WorkerKey] = Map.empty[ActorRef, WorkerKey]
 
   override def receive: Receive = {
-    case RegisterProjection(streamName, projectionName, workerName) =>
-      val metadata = WorkerMetadata(streamName, projectionName, workerName)
-      // when registering a projection worker, we default to state==enabled
-      val writeMajority = WriteMajority(timeout = 5.seconds)
-      replicator ! Update(DataKey, PNCounterMap.empty[WorkerMetadata], writeMajority)(
-        //TODO: read the default state from a desired _initial state_
-        // https://github.com/lagom/lagom/issues/1744
-        _.increment(node, metadata, 1)
-      )
-      // keep track and watch
-      actorIndex = actorIndex.updated(metadata, sender)
-      actorReverseIndex = actorReverseIndex.updated(sender, metadata)
+
+    case RegisterProjectionWorker(coordinates) =>
+      log.debug(s"Registering worker $coordinates to [${sender().path.toString}]")
+      // keep track
+      val workerKey: WorkerKey = coordinates.asKey
+      updateLWWMapForNameIndex(workerKey, coordinates)
+      actorIndex = actorIndex.updated(workerKey, sender())
+      reversedActorIndex = reversedActorIndex.updated(sender, workerKey)
+      // when worker registers, we must reply with the requested status (if it's been set already, or Started if not).
+      // TODO: parameterize the default initial status
+      val requestedStatus = requestedStatusLocalCopy.getOrElse(workerKey, Started)
+      log.debug(s"Setting requested status [$requestedStatus] on worker $workerKey [${sender().path.toString}]")
+      sender ! requestedStatus
+
+      // watch
       context.watch(sender)
 
-    case GetDesiredState =>
-      replicator ! Get(DataKey, ReadLocal, Some(sender()))
+    case GetState =>
+      sender ! State.fromReplicatedData(nameIndexLocalCopy, requestedStatusLocalCopy, observedStatusLocalCopy)
+
+    // StateRequestCommand come from `ProjectionRegistry` and contain a requested Status
+    case command: StateRequestCommand =>
+      // locate the target actor and send the request
+      log.debug(s"Propagating request $command.")
+      updateLWWMapForRequests(command.coordinates.asKey, command.requestedStatus)
```

Comment:
Any reason why we don't apply the changes to the worker when we get a request? 

If we can identify that the worker lives on this node, we can immediately apply the change without waiting to receive it via the replicator.

Reply:
I think it was safer but I can't remember the case I think the implementation where `ProjectionRegistry` would await for a response from `StateReuestCommand` demanded the change be propagated (ack'd by the majority) before sending an ACK back.